### PR TITLE
Remove legacy Hive readers and writers

### PR DESCRIPTION
**Pull request by electrum**
## Description

This PR will be merged once we have confidence in the new readers and writers. 

## Release notes

(x) Release notes are required, with the following suggested text:

```markdown
# Hive
* Remove legacy Hive readers. The `*_native_reader_enabled` session properties and the
  `*.native-reader.enabled` configuration properties have been removed. ({issue}`18241`)
* Remove legacy Hive writers. The `*_native_writer_enabled` session properties and the
  `*.native-writer.enabled` configuration properties have been removed. ({issue}`18241`)
* Remove support for S3 Select. The `s3_select_pushdown_enabled` session property
  and the `hive.s3select*` configuration properties have been removed. ({issue}`18241`)
* Optimized symlink listing is always enabled. The `optimize_symlink_listing` session property
  and `hive.optimize-symlink-listing` configuration property have been removed. ({issue}`18241`)
```

### CODE REVIEW

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestParquetPageSkippingWithOptimizedReader.java [hunk 1]
```diff
@@ -23,11 +24,13 @@ public class TestParquetPageSkippingWithOptimizedReader
     protected QueryRunner createQueryRunner()
             throws Exception
     {
+        if (true) {
+            // TODO (https://github.com/trinodb/trino/issues/9359)
+            throw new SkipException("Disabled until page level column indexes are supported in writer");
```
[the remark this answers is not in the document]
Comment (electrum) [reply plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestParquetPageSkippingWithOptimizedReader.java:29]: Could you put up a PR for this? We can start testing against such files now.

[the remark this answers is not in the document]
Comment (electrum) [reply plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestParquetPageSkippingWithOptimizedReader.java:29]: @raunaqmorarka Could you put up a PR for this? We can start testing against such files now.

**Review by dain — APPROVED** [review]

**Comment by dain** [comment]
This PR is missing release notes

**Comment by awakeljw** [comment]
Ihave two questions， Will the s3 select pushdown feature come back ？ Will s3 select pushdown feature support parquet file pushdown?

**Comment by electrum** [comment]
@awakeljw We have no plans to reimplement S3 Select. For Parquet, the results are returned as CSV or JSON, which is why it was never implemented in Trino. The feature introduces significant complexity in Trino and only benefits textfile formats. Additionally, the pricing model for S3 Select seems to put it at a disadvantage compared to simply increasing compute on the Trino side. Thus, we don't believe it is worth the complexity and maintenance burdon.

**Pull request merged**